src/test/java/net/snowflake/client/jdbc/ResultSetLatestIT.java

```diff
@@ -971,4 +979,138 @@ public void testLargeStringRetrieval() throws SQLException {
+          assertEquals("class java.math.BigDecimal", rs.getObject(2).getClass().toString());
+          assertEquals("class java.lang.Long", rs.getObject(3).getClass().toString());
+          assertEquals("class java.lang.Long", rs.getObject(4).getClass().toString());
+        }
+      }
+
+      // Test with jdbc_arrow_treat_decimal_as_int=false and JDBC_TREAT_DECIMAL_AS_INT=false
+      stmt.execute(setJdbcTreatDecimalAsIntFalse);
+      try (ResultSet rs = stmt.executeQuery(selectQuery)) {
+        while (rs.next()) {
+          assertEquals("class java.math.BigDecimal", rs.getObject(1).getClass().toString());
+          assertEquals("class java.math.BigDecimal", rs.getObject(2).getClass().toString());
+          assertEquals("class java.math.BigDecimal", rs.getObject(3).getClass().toString());
+          assertEquals("class java.math.BigDecimal", rs.getObject(4).getClass().toString());
+        }
+      }
+    }
+  }
+
+  // Test default setting of new connection property jdbc_arrow_treat_decimal_as_int=true. Connection property introduced after version 3.15.0.
+  @Test
+  public void testGetObjectForArrowResultFormatJDBCArrowDecimalAsIntTrue() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+         Statement stmt = con.createStatement()) {
+      stmt.execute("alter session set jdbc_query_result_format = 'ARROW'");
+      stmt.execute(createTableSql);
+      stmt.execute(insertStmt);
+
+      // Test with jdbc_arrow_treat_decimal_as_int=true and JDBC_TREAT_DECIMAL_AS_INT=true
+      try (ResultSet rs = stmt.executeQuery(selectQuery)) {
+        while(rs.next()) {
+          assertEquals("class java.lang.Long", rs.getObject(1).getClass().toString());
+          assertEquals("class java.math.BigDecimal", rs.getObject(2).getClass().toString());
+          assertEquals("class java.lang.Long", rs.getObject(3).getClass().toString());
+          assertEquals("class java.lang.Long", rs.getObject(4).getClass().toString());
+        }
+      }
+
+      // Test with jdbc_arrow_treat_decimal_as_int=true and JDBC_TREAT_DECIMAL_AS_INT=false
+      stmt.execute(setJdbcTreatDecimalAsIntFalse);
+      try (ResultSet rs = stmt.executeQuery(selectQuery)) {
+        while (rs.next()) {
+          assertEquals("class java.lang.Long", rs.getObject(1).getClass().toString());
+          assertEquals("class java.math.BigDecimal", rs.getObject(2).getClass().toString());
+          assertEquals("class java.lang.Long", rs.getObject(3).getClass().toString());
+          assertEquals("class java.lang.Long", rs.getObject(4).getClass().toString());
+        }
+      }
+    }
+  }
+
+  // Test setting new connection property jdbc_arrow_treat_decimal_as_int=false. Connection property
+  // introduced after version 3.15.0.
+  @Test
+  public void testGetObjectForJSONResultFormatJDBCArrowDecimalAsIntFalse() throws SQLException {
+    Properties properties = new Properties();
+    properties.put("jdbc_arrow_treat_decimal_as_int", false);
```

Comment:
we should not set this parameter here and the behaviour should be the same for JSON (but the whole test is ok to be added - needs only to be renamed)